services/dy-static-file-server/src/dy_static_file_server/inputs_to_outputs.py

```diff
@@ -112,8 +72,72 @@ def remap_input_to_output(input_dir: Path, output_dir: Path) -> None:
         json.dumps(outputs_key_values, sort_keys=True, indent=4)
     )
 
+    _logger.info("inputs to outputs sync successful!")
+
+
+class PortsMonitor:
+    def __init__(
+        self, input_dir: Path, output_dir: Path, *, monitor_interval: float = DEFAULT_MONITOR_WAIT_INTERVAL
+    ) -> None:
+        self.input_dir: Path = input_dir
+        self.output_dir: Path = output_dir
+        self.paths: set[Path] = {input_dir, output_dir}
+        self.monitor_interval: float = monitor_interval
+
+        self._monitor_task: asyncio.Task | None = None
+        self._keep_running: bool = False
+
+    def _get_state(self) -> Dict[Path, Dict[Path, Tuple[str, float]]]:
+        """return aggravated state for all monitored paths"""
+        return {p: _get_directory_state(p) for p in self.paths}
 
-def get_path_from_env(env_var_name: str) -> Path:
+    async def _monitor(self) -> None:
+
+        _logger.info("Started monitor")
+        previous_state = self._get_state()
+
+        while self._keep_running:
+            await asyncio.sleep(self.monitor_interval)
+
+            _logger.info("Checking")
+            current_state = self._get_state()
+
+            if previous_state != current_state:
+                _logger.info("Change detected!")
+                await self.on_change()
+
+            previous_state = current_state
+
+        _logger.info("Stopped monitor")
+
+    async def on_change(self) -> None:
+        try:
+            remap_input_to_output(self.input_dir, self.output_dir)
+            # alway regenerate index
+            generate_index()
+            _logger.info("on_change completed")
+        except Exception as e:
+            _logger.error("Could not finish remap of inputs to outputs %s", e)
+
+    async def start(self) -> None:
+        self._keep_running = True
+        self._monitor_task = asyncio.create_task(self._monitor())
+
+    async def stop(self) -> None:
+        if self._monitor_task:
+            self._keep_running = False
+            await self._monitor_task
+            self._monitor_task = None
+
+
+async def _run_background_sync(input_dir: Path, output_dir: Path) -> None:
+    monitor = PortsMonitor(input_dir, output_dir)
+    await monitor.start()
+    while True:
+        await asyncio.sleep(1)
```

Comment:
Don't you need to add calls to the `monitor` here?

Reply:
No, it's supposed to wait her forever.